parcels/codegenerator.py

```diff
@@ -152,7 +152,7 @@ def __getitem__(self, attr):
 
 
 class MathNode(IntrinsicNode):
-    symbol_map = {'pi': 'M_PI', 'e': 'M_E'}
+    symbol_map = {'pi': 'M_PI', 'e': 'M_E', 'nan': 'NAN'}
```

Comment:
agreed - I didn't even know that the NAN symbol exists in math.h, but it appears to be standard.